src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.AspNetCore.App.Analyzers.Infrastructure;
+using Microsoft.CodeAnalysis;
+using WellKnownType = Microsoft.AspNetCore.App.Analyzers.Infrastructure.WellKnownTypeData.WellKnownType;
+
+namespace Microsoft.AspNetCore.Http.Generators.StaticRouteHandlerModel;
+
+internal class EndpointParameter
+{
+    public EndpointParameter(IParameterSymbol parameter, WellKnownTypes wellKnownTypes)
+    {
+        Type = parameter.Type;
+        Name = parameter.Name;
+        Source = EndpointParameterSource.Unknown;
+
+        if (GetSpecialTypeCallingCode(Type, wellKnownTypes) is string callingCode)
+        {
+            Source = EndpointParameterSource.SpecialType;
+            CallingCode = callingCode;
+        }
+    }
+
+    public ITypeSymbol Type { get; }
+    public EndpointParameterSource Source { get; }
+
+    // TODO: If the parameter has [FromRoute("AnotherName")] or similar, prefer that.
+    public string Name { get; }
+    public string? CallingCode { get; }
+
+    public string EmitArgument()
+    {
+        switch (Source)
+        {
+            case EndpointParameterSource.SpecialType:
+                return CallingCode!;
+            default:
+                // Eventually there should be know unknown parameter sources, but in the meantime we don't expect them to get this far.
+                // The netstandard2.0 target means there is no UnreachableException.
+                throw new Exception("Unreachable!");
+        }
+    }
+
+    // TODO: Handle special form types like IFormFileCollection that need special body-reading logic.
```

Comment:
Just for curiosity, what is your idea about form types? Probably you'll need to track them to call `ReadFormAsync` somewhere, right?

Reply:
Yeah. It will be similar to JSON bodies, BindAsync, and even TryParse handling.